src/lib/components/references/bibleTextReferenceFetcher.ts

```diff
@@ -180,7 +186,11 @@ export const fetchAndFormat = async (
                 return null;
             }
 
-            verseDisplayName = generateVerseRangeDisplayName(passageBookTexts, language, false);
+            // handle the case where verse 0 was requested in addition to only requesting verse 1 but verse 0 doesn't exist in the target Bible
+            verseDisplayName =
+                passageBookTexts.length === 1
```

Comment:
🤔 Maybe it would be better to have a single function that does this check and generates the correct reference based upon `passageBookTexts.length` instead of two different functions, `generateSingleVerseDisplayName` and `generateVerseRangeDisplayName`?

Reply:
Yeah, this check is wrong anyway because it's only checking to see if there's only one book.  I'll update to use a single function.